### PR TITLE
Feature/dyanmic config loading for cli and light

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

File: libraries/chain/include/sophiatx/chain/genesis_state.hpp [hunk 1]
```diff
@@ -31,47 +31,154 @@
...
+        account_name_type name;
+        public_key_type key;
+        share_type balance;
+    };
+
+    public_key_type initial_public_key = sophiatx::protocol::public_key_type(SOPHIATX_INIT_PUBLIC_KEY_STR);
+
+    public_key_type initial_public_mining_key = sophiatx::protocol::public_key_type(SOPHIATX_INIT_PUBLIC_KEY_STR);
+#undef SOPHIATX_INIT_PUBLIC_KEY_STR
+
```
Comment (JakubFornadel) [inline libraries/chain/include/sophiatx/chain/genesis_state.hpp]: might be nicer to use default value directly here, no #define and #undef is then needed

Comment (MatusKysel) [reply libraries/chain/include/sophiatx/chain/genesis_state.hpp]: Yes, I agree, but then there will be some defines in config.hpp and some in here, so it is more confusing

Comment (JakubFornadel) [reply libraries/chain/include/sophiatx/chain/genesis_state.hpp]: how about making all the newly added parameters in genesis optional ? Then genesis state does not have to have default values(default genesis.txt stays the same as now or can optionally include those new params), we can delete all the macros and default values will be in init function(or constructor) of config class and only those genesis parameters that have some value will be overridden in config

**Review by JakubFornadel — APPROVED** [review]

File: libraries/chain/include/sophiatx/chain/get_config.hpp [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+#include <sophiatx/protocol/config.hpp>
+#include <sophiatx/protocol/protocol_config.hpp>
+#include <sophiatx/protocol/asset.hpp>
+#include <sophiatx/protocol/types.hpp>
+#include <sophiatx/protocol/version.hpp>
+#include <sophiatx/chain/genesis_state.hpp>
+
+namespace sophiatx { namespace chain {
+
+class sophiatx_config {
```
Comment (JakubFornadel) [inline libraries/chain/include/sophiatx/chain/get_config.hpp:12]: if it was class siophiatx_config : private fc::mutable_variant_object {
...
}
you would not need class member config_

**Pull request merged**